library/src/main/scala/scala/scalajs/js/JSConverters.scala

```diff
@@ -41,6 +42,46 @@ object JSConverters extends JSConvertersLowPrioImplicits {
     @inline final def toJSArray: Array[T] = genTraversableOnce2jsArray(col)
   }
 
+  implicit class JSRichGenIterable[T](val col: GenIterable[T]) extends AnyVal {
```

Comment:
Use obscure names like `__self` for the underlying field of an `implicit` value class, so that it's obvious it should not be used directly on a `GenIterable`.